libs/blocks/caas/utils.js

```diff
@@ -523,6 +523,19 @@ export function getCountryAndLang({ autoCountryLang, country, language }) {
     const pathArr = pageConfigHelper()?.pathname?.split('/') || [];
     const langStr = LANGS[pathArr[1]] ?? LANGS[''] ?? 'en';
     let countryStr = LOCALES[pathArr[2]] ?? 'xx';
+
+    // If no country found in URL path, try GEO IP lookup
+    if (countryStr === 'xx') {
```

Comment:
Is the 'xx' approach necessary? 🤔
```suggestion
    let countryStr = LOCALES[pathArr[2]];

    // If no country found in URL path, try GEO IP lookup
    if (countryStr) {
```